util/csvresponse.py

```diff
@@ -86,14 +88,31 @@ def _create_filter_list(self,keys):
         # start with fields we never want to include
         drop = {'id', 'annotations'}
         for key in keys:
-            # remove any "extra" keys while keeping qc params and removing 'provenance' params
-            if (key not in params_to_include and not self._is_qc_param(key)) or 'provenance' in key:
+            # remove any "extra" keys while keeping relevant qc params and removing 'provenance' params
+            if self._is_qc_param(key):
+                # only include if a requested param matches the QC Key
+                if not self._needs_qc_param(key, params_to_include):
```

Comment:
`_needs_qc_param` is only used once. I don't think it's worth the overhead of a function call. Also `_needs_qc_param` is needlessly using the regular expression library for a built-in string function `split`. Consider this:

```python
    if self._is_qc_param(key):
        # only include if a requested param matches the QC Key
        if key.split('_qc_')[0] in params_to_include:
            drop.add(key)
```

Reply:
Agree.